contracts/PoolManager.sol

```diff
@@ -251,11 +253,11 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC1155, IERC1155Rec
             }
         }
 
+        PoolId id = key.toId();
         // Set the total swap fee, either through the hook or as the static fee set an initialization.
         uint24 totalSwapFee;
         if (key.fee.isDynamicFee()) {
-            totalSwapFee = IDynamicFeeManager(address(key.hooks)).getFee(msg.sender, key, params, hookData);
-            if (totalSwapFee >= 1000000) revert FeeTooLarge();
+            totalSwapFee = pools[id].slot0.dynamicFee;
         } else {
             // clear the top 4 bits since they may be flagged for hook fees
             totalSwapFee = key.fee.getStaticFee();
```

Comment:
cleanup idea -- now that the dynamic fee is stored in slot0, can we just also store the static fee in slot0? That would unify this branching and in Pool.initialize we could pass in either the static fee or current dynamic fee

This is kinda nice as then the "source of truth" for the current swap fee is in a single place in slot0 and the dynamism is just from updating that value via hook if allowed

Reply:
note this is only worth it if the slot0 reading is 0 cost (which in theory it should be since we must fetch from slot0 for every swap anyways)